docs/admin/catalog.md

```diff
@@ -0,0 +1,92 @@
+
+This section will describe the structure of the Catalog, and how to extend it if needed.
+
+Catalog Entries
+---------------
+
+The Catalog itself is a repository containing one to many entries. Each entry is a folder, named after the Service Type the entry represents, which contains as a minimum requirement:
+
+  - A docker-compose.yml file which describes the deployment of the IoT Service as a Docker Stack on Docker Swarm
+  - A mastermind.yml file, which describes the Service and its configuration options. This is what MasterMind uses to generate the UI forms to create and deploy new Services.
+
+In addition, depending on the contents of the docker compose file, each entry can also come with:
+
+  - Config files, to be used to create Docker configs
+  - Secret files, to be used to create Docker secrets
+
+Deployment Template
+-------------------
+
+The deployment template for a Catalog entry comes in the form of a docker-compose.yml file, which describes how to deploy the Service on a Docker Swarm cluster. These files should be compliant with Docker Compose Version 3 in order to be used by MasterMind, however not all options available in Compose are currently supported. For a list of supported options see *LINK NEEDED*
+
+To provide MasterMind users with some degree of configuration at deployment time, the docker-compose.yml file can define environment variable substitution as such:
+
+```
+environment:
+  - CONFIG_VARIABLE=${CONFIG_VARIABLE}
+```
+
+These can then be defined in the configuration template, to be provided by the user through MasterMind's UI when creating a new Service.
+
+Configuration Template
+----------------------
+
+The configuration template for a Catalog entry comes in the form of a YAML file, named mastermind.yml. This file follows a specific structure, and defines metadata regarding the Services this entry can deploy, as well as how the Services are configurable at deployment time.
+
+An example of a (mock) mastermind.yml file:
+
+```
+name: comet
+version: 1.0.0
+description: Short Time Historic Data Management Service
+protocol_type: HTTP
+ngsi_version: 1
+environment_variables:
+  - variable: COMET_VERSION
+    name: Comet Version
+    description: The version of STH Comet to use
+    required: true
+    managed: true
+    default: latest
+  - variable: REPLICASET_NAME
+    name: Replicaset Name
+    description: The name of the Replica Set to use
+    required: true
+    managed: true
+    default: rs
+services:
+  - service_type: mongo
+    name: Mongo
+    description: Mongo database to connect to
+    required: true
+    managed: true
+    retrieve: endpoint
+    as: MONGO_SERVICE_URI
+```
+
+The following options should be defined in the mastermind.yml file:
+
+  - name: the name of the Service Type (e.g. orion)
+  - version: the version of this Service Type (e.g. 1.0.3)
```

Comment:
@chicco785 I think I coded it so that there could be different top level folders per version (Orion-1.0, Orion-1.2, ...), though I should probably do it as you suggested instead, it's easier to organise.